summoner-tui/src/Summoner/Tui/Validation.hs

```diff
@@ -189,6 +193,14 @@ validateKit dirs kit =
         (not $ kit ^. projectMeta . lib  || kit ^. projectMeta . exe)
         LibOrExe
 
+    validatePreludePackage :: Validation (NonEmpty FormError) ()
+    validatePreludePackage =
+      let packageName = kit ^. projectMeta . preludeName in
+      let isPreludePackageInvalid = not $ T.null packageName || packageNameValid packageName in
```

Comment:
Why is there a check on `null packageName`? It is a valid option to not specify the custom prelude in the current design.

Reply:
That predicate is negated, we simply allow empty and well-formatted package names here. 
`packageNameValid` function is a general-purpose check for validity of a given package name, as per Cabal spec. Since empty package names are not in general valid, but only for our application, I've localized null check here.

Reply:
Oh, yeah, sorry, it's my mistake, didn't look carefully!